web/docs/tql2/operators/splunk.md

```diff
@@ -0,0 +1,108 @@
+# splunk
+
+Sends events to a Splunk [HTTP Event Collector (HEC)][hec].
+
+[hec]: https://docs.splunk.com/Documentation/Splunk/9.3.1/Data/UsetheHTTPEventCollector
+
+```tql
+splunk url:str, hec_token=str,
+      [host=str, source=str, sourcetype=expr, index=expr,
+       tls_no_verify=bool, print_nulls=bool,
+       max_content_length=int, send_timeout=duration, compress=bool]
+```
+
+## Description
+
+The `splunk` operator sends events to a Splunk [HTTP Event Collector
+(HEC)][hec].
+
+The source type defaults to `_json` and the operator renders incoming events as
+JSON. You can specify a different source type via the `sourcetype` option.
+
+The operator accumulates multiple events before sending them as a single
+message to the HEC endpoint. You can control the maximum message size via the
+`max_content_length` and the timeout before sending all accumulated events via
+the `send_timeout` option.
+
+### `url: str`
+
+The address of the Splunk indexer.
+
+### `hec_token = str`
+
+The [HEC
+token](https://docs.splunk.com/Documentation/Splunk/9.3.1/Data/UsetheHTTPEventCollector#Create_an_Event_Collector_token_on_Splunk_Cloud_Platform)
+for authentication.
+
+### `host = str (optional)`
+
+An optional value for the [Splunk `host`](https://docs.splunk.com/Splexicon:Host).
+
+### `source = str (optional)`
+
+An optional value for the [Splunk `source`](https://docs.splunk.com/Splexicon:Source).
+
+### `sourcetype = expr (optional)`
+
+An optional expression for [Splunk's
+`sourcetype`](https://docs.splunk.com/Splexicon:Sourcetype) that evaluates to a
+`string`. You can use this to set the `sourcetype` per event.
+
+Regardless of the chosen `sourcetype`, the event itself is given as the value to
+the `event` key in the top level object being send.
+
+Defaults to `_json`.
```

Comment:
Maybe link to https://docs.splunk.com/Documentation/Splunk/9.3.1/Data/Listofpretrainedsourcetypes as list of possible sourcetypes.

Reply:
I just thought about this some more, and I now think that we maybe *shouldnt* do that, as we dont actually support those source types as-is. We always write the entire body into the top-level `event` key, which isnt correct for more of the predefined ones.